MatomoTracker/MatomoTracker.swift

```diff
@@ -399,6 +399,25 @@ extension MatomoTracker {
             dimension.index != index
         })
     }
+    
+    /// remove all without ignore
+    @objc public func removeAll(ignoreIndexs indexs: [NSNumber]? = nil) {
+        if let array = indexs, array.count > 0 {
+            
+            let tempArray = dimensions
+            dimensions.removeAll()
+            for i in array {
+                let resultArray  = tempArray.filter({ (dimension) -> Bool in
+                    dimension.index == i.intValue
+                })
+                if resultArray.count > 0, let customDimension = resultArray.first {
+                    dimensions.append(customDimension)
+                }
+            }
+        } else {
+            dimensions.removeAll()
+        }
+    }
```

Comment:
One possible implementation would be

````
public func removeAll(ignoreIndexs indexes: Set<Int>? = nil) {

    guard let indexes = indexes else {
        dimensions.removeAll()
        return
    }

    let filtered = dimensions.filter { indexes.contains($0.index) }
    dimensions = filtered

}
````
(might be some typos here, have not compiled it)